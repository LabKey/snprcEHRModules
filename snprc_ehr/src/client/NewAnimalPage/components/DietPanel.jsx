import React from 'react';
import Select from 'react-select';
import moment from 'moment';
import WrappedDatePicker from '../../utils/components/WrappedDatePicker';
import InfoPanel from './InfoPanel';

export default class DietPanel extends React.Component {
    componentDidMount = () => {
        this.props.preventNext(); //prevent/Allow Next button
    }
    handleDateSelect = date => {
        //do nothing
    };
    handleDateChangeRaw = e => {
        e.preventDefault();
    }
    handleDataChange = option => {
        this.props.handleDataChange('diet', option);
    }
    render() {
        let { diet, acqDate } = this.props.newAnimalData;
        return (
            <>
                <div className="wizard-panel__rows">
                    <div className='wizard-panel__row' >
                        <div className='wizard-panel__col'>
                            <div className="diet-datepicker">
                                <WrappedDatePicker
                                    label="Diet Date"
                                    todayButton="Today"
                                    dateFormat="Pp"
                                    selected={acqDate.date.toDate()}
                                    onSelect={this.handleDateSelect}
                                    onChange={this.handleDateSelect}
                                    onChangeRaw={this.handleDateChangeRaw}
                                    readOnly={true}
                                    disabledKeyboardNavigation={true}
                                    disabled={this.props.disabled}
                                />
                            </div>
                        </div>
                    </div>
                    <div className='wizard-panel__row' >
                        <div className='wizard-panel__col'>
                            <label className="field-label" >Diet</label>
                            <Select
                                defaultValue={diet}
                                className="shared-dropdown"
                                classNamePrefix="shared-select"
                                options={this.props.dietList}
                                onChange={this.handleDataChange}
                                placeholder="Select diet"
                                isDisabled={this.props.disabled}
                                isClearable={true}
                                isLoading={this.props.dietList.length === 0}
                                autoFocus
                            />
                        </div>
                    </div>
                </div>
                <InfoPanel
                    messages={[{ propTest: !diet, colName: 'Diet' }]}
                />
            </>
        )
    }
}
