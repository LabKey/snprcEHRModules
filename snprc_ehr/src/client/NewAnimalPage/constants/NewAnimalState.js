import moment from 'moment';

const NewAnimalState = () => {

    return  ({
        currentPanel: 1,
        selectedOption: undefined,
        newAnimalData: {
            selectedOption: undefined,
            id: undefined,
            birthDate: { date: moment() },
            bdStatus: undefined,
            acquisitionType: undefined,
            acqDate: { date: moment() },
            gender: undefined,
            sire: undefined,
            dam: undefined,
            species: undefined,
            colony: undefined,
            animalAccount: undefined,
            ownerInstitution: undefined,
            responsibleInstitution: undefined,
            room: undefined,
            cage: { value: undefined },
            diet: undefined,
            pedigree: undefined,
            iacuc: undefined
        },
        summaryData: [],
        speciesList: [],
        acquisitionTypeList: [],
        potentialDamList: [],
        potentialSireList: [],
        locationList: [],
        accountList: [],
        colonyList: [],
        dietList: [],
        institutionList: [],
        iacucList: [],
        pedigreeList: [],
        bdStatusList: [],
        isLoading: true,
        hasError: false,
        preventNext: true,
        saveOk: false,
        showSaveModal: false,
        showCancelModal: false,
        showSpeciesChangeModal: false,
        errorMessage: undefined
    });
}

export default NewAnimalState;
