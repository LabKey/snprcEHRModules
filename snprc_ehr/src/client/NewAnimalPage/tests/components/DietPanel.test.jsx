import React from 'react'
import { shallow } from 'enzyme'
import DietPanel from '../../components/DietPanel'
import NewAnimalState from '../../constants/NewAnimalState'
import lists from '../fixtures/lists'

jest.mock('moment', () => {
    return () => jest.requireActual('moment')('2023-01-01T00:00:00.000Z');
});

test('Should render the DietPanel', () => {
    const { newAnimalData } = new NewAnimalState()

    const wrapper = shallow(
      <DietPanel
        dietList={ lists.dietList }
        disabled={ false }
        handleDataChange={ () => { return false } }
        newAnimalData={ newAnimalData }
        preventNext={ () => { return false } }
      />
    )
    expect(wrapper).toMatchSnapshot()
})
