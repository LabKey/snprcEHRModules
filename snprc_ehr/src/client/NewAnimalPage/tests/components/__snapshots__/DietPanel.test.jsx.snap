// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the DietPanel 1`] = `
<Fragment>
  <div
    className="wizard-panel__rows"
  >
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div
          className="diet-datepicker"
        >
          <WrappedDatePicker
            dateFormat="Pp"
            disabled={false}
            disabledKeyboardNavigation={true}
            label="Diet Date"
            onChange={[Function]}
            onChangeRaw={[Function]}
            onSelect={[Function]}
            readOnly={true}
            selected={2023-01-01T00:00:00.000Z}
            todayButton="Today"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Diet
        </label>
        <ForwardRef
          autoFocus={true}
          className="shared-dropdown"
          classNamePrefix="shared-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            [
              {
                "id": 0,
                "label": "Regular chow",
                "species": null,
                "value": 1,
              },
              {
                "id": 1,
                "label": "Corn feed",
                "species": null,
                "value": 2,
              },
            ]
          }
          placeholder="Select diet"
        />
      </div>
    </div>
  </div>
  <InfoPanel
    messages={
      [
        {
          "colName": "Diet",
          "propTest": true,
        },
      ]
    }
  />
</Fragment>
`;
